cpp/client/src/main.cpp

```diff
@@ -1,159 +1,178 @@
+        LogError("") << "loading params from " << configFile << " met error!" << std::endl;
+        return;
+    }
+
+    LogInfo("") << "All headlessParams:" << headlessParams << std::endl;
+    LogDebug("") << "general configuration:\n" << headlessParams.generalParams_ << std::endl;
+    LogDebug("") << "audioConsumerParams configuration:\n" << headlessParams.audioConsumerParams_ << std::endl;
+    LogDebug("") << "videoConsumerParams configuration:\n" << headlessParams.videoConsumerParams_ << std::endl;
+
+    // setup audio fetching
+    const int audioStreamsNumber = headlessParams.defaultAudioStreams_.size();
+
+    for (int audioStreamsCount = 0; audioStreamsCount < audioStreamsNumber; audioStreamsCount++) {
+        MediaStreamParamsSupplement *audioStream = headlessParams.getMediaStream(headlessParams.defaultAudioStreams_, audioStreamsCount);
+        LogDebug("") << "initiating remote audio stream for "
+                     << audioStream->streamPrefix_ << std::endl;
+
+        std::string audioStreamPrefix = ndnp->addRemoteStream(audioStream->streamPrefix_ , audioStream->threadToFetch_, (*audioStream), headlessParams.generalParams_, headlessParams.audioConsumerParams_, NULL);
+        remoteStreamsPrefix.push_back(audioStreamPrefix);
+        LogInfo("") << "demo audio fetching " << audioStreamsCount << " from " << audioStreamPrefix << " initiated, "
+                    << "threadToFetch: " << audioStream->threadToFetch_ << std::endl;
+    }
+
+    // setup video fetching
+    const int videoStreamsNumber = headlessParams.defaultVideoStreams_.size();
+
+
+    RendererInternal *renderer = new RendererInternal[videoStreamsNumber];
+
+    for (int videoStreamsCount = 0; videoStreamsCount < videoStreamsNumber; videoStreamsCount++) {
+        MediaStreamParamsSupplement *videoStream = headlessParams.getMediaStream(headlessParams.defaultVideoStreams_, videoStreamsCount);
+
+        LogDebug("") << "initiating remote video stream for " << videoStream->streamPrefix_ << ", vconsumerParams: " << headlessParams.videoConsumerParams_ << std::endl;
+        std::string videoStreamPrefix = ndnp->addRemoteStream(videoStream->streamPrefix_, videoStream->threadToFetch_, (*videoStream), headlessParams.generalParams_, headlessParams.videoConsumerParams_, &renderer[videoStreamsCount]);
+        remoteStreamsPrefix.push_back(videoStreamPrefix);
+        LogInfo("") << "demo video fetching " << videoStreamsCount << " from " << videoStreamPrefix << " initiated, "
+                    << "threadToFetch: " << videoStream->threadToFetch_ << std::endl;
+    }
+
+
+    // collect streams statictics
+    boost::asio::io_service staticticsIo;
+    LogDebug("") << "statisticsSampleInterval(s): " << statisticsSampleInterval << std::endl;
+    collectStreamsStatictics collectStatictics(staticticsIo, headlessAppOnlineTimeSec, statisticsSampleInterval, headlessParams.statistics_, remoteStreamsPrefix, ndnp);
```

Comment:
1/ let's think of it as a "StatPoller" or "StatCollector"
2/ no need to provide `headlessAppOnlineTimeSec`